src/ir/auth_logic/souffle_emitter.h

```diff
@@ -102,14 +120,48 @@ class SouffleEmitter {
     return absl::StrCat(".decl ", predicate.name(), "(", arguments, ")");
   }
 
-  std::string EmitDeclarations() {
-    std::vector elements_vec(std::make_move_iterator(declarations_.begin()),
-                             std::make_move_iterator(declarations_.end()));
-    std::sort(elements_vec.begin(), elements_vec.end());
-    return absl::StrJoin(elements_vec, "\n",
-                         [this](std::string* out, auto decl) {
-                           return absl::StrAppend(out, EmitDeclaration(decl));
-                         });
+  std::string EmitArguments(std::vector<datalog::Argument> arguments) {
+    return absl::StrJoin(
+        arguments, ", ", [](std::string* out, datalog::Argument argument) {
+          return absl::StrAppend(out, argument.argument_name(), " : ",
+                                 argument.argument_type().name());
+        });
+  }
+
+  std::string EmitRelationDeclarations(
+      const datalog::Program& program,
+      const std::vector<std::string>& skip_declarations) {
+    std::vector<std::string> declaration_strings;
+    for (const auto& declaration : program.relation_declarations()) {
+      if (absl::c_find(skip_declarations, declaration.relation_name()) !=
+          skip_declarations.end())
```

Comment:
This `c_find` is `O(N)`, where `N` is the number of `skip_declarations`. It could instead be `O(1)` if `skip_declarations` were instead an `absl::flat_hash_set` of `std::string`.

Reply:
Ah, I thought `skip_declarations` was a `set`

Reply:
Ahh I missed this optimization. Thanks for the suggestion!